components/org.wso2.micro.integrator.extensions/org.wso2.micro.integrator.management.apis/src/main/java/org/wso2/micro/integrator/management/apis/RequestCountResource.java

```diff
@@ -0,0 +1,142 @@
+
+        if (tryParseInt(yearParameter) != null && tryParseInt(monthParameter) != null) {
+            if (!StringUtils.isEmpty(yearParameter) && !StringUtils.isEmpty(monthParameter)) {
+                return takeRequestCuntOfTheMonth(synCtx, Integer.parseInt(yearParameter),
+                        Integer.parseInt(monthParameter));
+            } else if (StringUtils.isEmpty(yearParameter) && StringUtils.isEmpty(monthParameter)) {
+                Date date = new Date();
+                LocalDate localDate = date.toInstant().atZone(ZoneId.systemDefault()).toLocalDate();
+                return takeRequestCuntOfTheMonth(synCtx, localDate.getYear(), localDate.getMonthValue());
+            }
+        }
+        JSONObject response = Utils.createJsonError("Invalid Input", axisCtx, BAD_REQUEST);
+        axisCtx.removeProperty(org.wso2.micro.integrator.management.apis.Constants.NO_ENTITY_BODY);
+        Utils.setJsonPayLoad(axisCtx, response);
+        return true;
+    }
+
+    // Take aggregated request count for a given month.
+    private boolean takeRequestCuntOfTheMonth(MessageContext synCtx, int year, int month) {
+
+        JSONObject response;
+        org.apache.axis2.context.MessageContext axisCtx = ((Axis2MessageContext) synCtx).getAxis2MessageContext();
+        axisCtx.setProperty(org.apache.axis2.Constants.Configuration.MESSAGE_TYPE, "application/json");
+        axisCtx.setProperty(Constants.Configuration.CONTENT_TYPE, "application/json");
+        // Sending a response body for a GET request
+        axisCtx.removeProperty(NO_ENTITY_BODY);
+
+        TransactionStore transactionStore = DataHolder.getInstance().getTransactionStore();
+        if (transactionStore != null) {
+            RDBMSConnector rdbmsConnector = transactionStore.getRdbmsConnector();
+            // Get request count of current Month
+            long requestCount;
+            try {
+                requestCount = rdbmsConnector.getTransactionCountOfMonth(year, month);
+            } catch (SQLException e) {
+                response =
+                        Utils.createJsonError("Error occurred while retrieving data from the database", axisCtx,
+                                INTERNAL_SERVER_ERROR);
+                Utils.setJsonPayLoad(axisCtx, response);
+                log.error("Error occurred while retrieving data from the database", e);
+                return true;
+            }
+            if (requestCount != -1L) {
+                response = new JSONObject(
+                        "{\"Year\" : " + year + ", \"Month\" : " + month + ", \"RequestCount\" : " + requestCount +
+                                "}");
+            } else {
+                response = new JSONObject("{\"error\" : \"Did not find stats for " + year + "/" + month + "\"}");
+            }
+        } else {
+            response =
+                    Utils.createJsonError("TransactionStore is not initialized", axisCtx, INTERNAL_SERVER_ERROR);
```

Comment:
Is INTERNAL_SERVER_ERROR the correct status code? Wouldn't 403, forbidden suite better? because there's nothing wrong with the request

Reply:
+1 fixed